base/db/R/dbfiles.R

```diff
@@ -636,3 +636,63 @@ dbfile.id <- function(type, file, con, hostname=PEcAn.remote::fqdn()) {
     invisible(NA)
   }
 }
+
+#' Function to Insert Records into the Formats and Variables Tables
+#' @name dbfile.format.variable.insert
+#' @param format_name Format Name
+#' @param header TRUE/FALSE Boolean describing whether or not the file contains a header
+#' @param skip The number of lines in the header
+#' @param mimetype_id The mimetype id of the mimetype that is associated with the format
+#' @param format_notes Any useful notes that describe the format being created
+#' @param con Connection object to connect to bety
+#' @param format_variables A TRUE/FALSE Boolean used to toggle a format-variables db query
+#' @param description variable descripthon
+#' @param units variable units
+#' @param var_notes notes to further describe the variable
+#' @param var_name variable name
+#' @param max maximum possible value
+#' @param min minimum possible value
+#' @param standard_name PEcAn specific name? 
+#' @param label ?
+#' @param type ?
+#' @author Liam Burke
+#'
+#' @return formatid
+#' @export
+#' 
+#' @examples
+#' \dontrun{
+#' bety <- betyConnect()
+#'   dbfile.format.variable.insert('Test_Format', header = TRUE, skip = 2, mimetype_id = 1060, con = bety$con, format_variables = FALSE)
+#' }
+dbfile.format.variable.insert <- function(format_name, header, skip, mimetype_id, format_notes = "", con, format_variables = FALSE, 
+                                          description = NA, units = NA, var_notes = NA, var_name = NA, max = NA, min = NA, standard_name = NA, label = NA, type = NA){
+ 
+  formatid <- get.id(
```

Comment:
Really don't follow what you're doing here. It looks like you're using `get.id` to create the format record, but with only just the name and mimetype. Next, you use a second query to create a DIFFERENT format record, that has `header, skip, mimetype_id, notes, name`, but doesn't include the `returning id` necessary to get the format ID.   Therefore, this second record with more information is never returned. You need to make a choice between these two ways of creating records and do just one, not both.

Reply:
@mdietze  My mistake. I thought that `get.id` resolved the `format id` but I now see that this was a mistake. 

can you clarify what I need to do to return the format id from the second type of query?

Reply:
Never mind. I see this in the comments below. Thanks.